pkgs/development/python-modules/xtensor-python/default.nix

```diff
@@ -10,26 +10,37 @@
 
 buildPythonPackage rec {
   pname = "xtensor-python";
-  version = "0.25.1";
+  version = "0.26.1";
+
+  format = "other"; # Just CMake
 
   src = fetchFromGitHub {
     owner = "xtensor-stack";
     repo = pname;
     rev = version;
-    sha256 = "17la76hn4r1jv67dzz8x2pzl608r0mnvz854407mchlzj6rhsxza";
+    sha256 = "sha256-kLFt5Ah5/ZO6wfTZQviVXeIAVok+/F/XCwpgPSagOMo=";
   };
 
   nativeBuildInputs = [ cmake pybind11 ];
+  checkInputs = [ gtest ]; # find_package fails if we use nativeCheckInputs
+  nativeCheckInputs = [ gtest ];
+
+  cmakeFlags = [
+    "-DBUILD_TESTS=ON"
+  ];
 
   propagatedBuildInputs = [ xtensor numpy ];
 
   dontUseSetuptoolsBuild = true;
   dontUsePipInstall = true;
   dontUseSetuptoolsCheck = true;
 
-  nativeCheckInputs = [
-    gtest
-  ];
+  # NOTE: buildPythonPackage only accounts for preInstallCheck, not for
+  # preCheck
+  checkTarget = "xtest";
+  preInstallCheck = ''
+    make $checkTarget
+  '';
```

Comment:
We can add `pytestCheckHook` and `setuptools` to `nativeCheckInputs` and run tests under "test_python/".

Reply:
This hook runs the C++ tests/doctest, not pytest. But you're right they seem to also have python tests, which this derivation currently doesn't seem to run. I should look into this

Reply:
> This hook runs the C++ tests/doctest, not pytest. But you're right they seem to also have python tests, which this derivation currently doesn't seem to run. I should look into this

Yes I agree.